packages/@aws-cdk/aws-autoscaling/test/test.auto-scaling-group.ts

```diff
@@ -1088,24 +1088,136 @@ export = {
-        instanceType: ec2.InstanceType.of(ec2.InstanceClass.M4, ec2.InstanceSize.MICRO),
-        machineImage: new ec2.AmazonLinuxImage(),
-        vpc,
-        notificationsTopic: topic,
-        notifications: [{
-          topic,
-        }],
-      });
-    }, 'Can not set notificationsTopic and notifications, notificationsTopic is deprected use notifications instead');
+    // When
+    asg.emitAllGroupMetrics();
+
+    // Then
+    expect(stack).to(haveResource('AWS::AutoScaling::AutoScalingGroup', {
+      MetricsCollection: [
+        {
+          'Granularity': '1Minute',
+          Metrics: ABSENT,
+        },
+      ],
+    }));
+    test.done();
+  },
+
+  'test configuring metricsCollection to GroupMetric.ALL adds a single MetricsCollection with no Metrics specified'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = mockVpc(stack);
+    // When
+    new autoscaling.AutoScalingGroup(stack, 'ASG', {
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.M4, ec2.InstanceSize.MICRO),
+      machineImage: new ec2.AmazonLinuxImage(),
+      vpc,
+      groupMetricsCollections: [
+        {
+          metrics: autoscaling.GroupMetric.ALL,
+        },
+      ],
+    });
+
+    // Then
+    expect(stack).to(haveResource('AWS::AutoScaling::AutoScalingGroup', {
+      MetricsCollection: [
+        {
+          Granularity: '1Minute',
+          Metrics: ABSENT,
+        },
+      ],
+    }));
+    test.done();
+  },
+
+  'test emitGroupMetrics adds MetricsCollection with a list of specified Metrics'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = mockVpc(stack);
+    const asg = new autoscaling.AutoScalingGroup(stack, 'ASG', {
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.M4, ec2.InstanceSize.MICRO),
+      machineImage: new ec2.AmazonLinuxImage(),
+      vpc,
+    });
+
+    // When
+    asg.emitGroupMetrics(
+      autoscaling.GroupMetric.MIN_SIZE,
+      autoscaling.GroupMetric.MAX_SIZE,
+      autoscaling.GroupMetric.DESIRED_CAPACITY,
+      autoscaling.GroupMetric.IN_SERVICE_INSTANCES,
+    );
+    asg.emitGroupMetrics(
+      autoscaling.GroupMetric.PENDING_INSTANCES,
+      autoscaling.GroupMetric.STANDBY_INSTANCES,
+      autoscaling.GroupMetric.TOTAL_INSTANCES,
+      autoscaling.GroupMetric.TERMINATING_INSTANCES,
+    );
+
+    // Then
+    expect(stack).to(haveResource('AWS::AutoScaling::AutoScalingGroup', {
+      MetricsCollection: [
+        {
+          Granularity: '1Minute',
+          Metrics : [ 'GroupMinSize', 'GroupMaxSize', 'GroupDesiredCapacity', 'GroupInServiceInstances' ],
+        }, {
+          Granularity: '1Minute',
+          Metrics : [ 'GroupPendingInstances', 'GroupStandbyInstances', 'GroupTotalInstances', 'GroupTerminatingInstances' ],
+        },
+      ],
+    }));
+    test.done();
+  },
+
+  'can configure metricsCollection on the constructor and using emitGroupMetrics simultaneously'(test: Test) {
```

Comment:
Does it work correctly if they are repeated, i.e., do we dedupe? Add a test case

Reply:
What do you mean?  This is dedupe in ASG , meaning it will not fail/generate the same group metrics in the same **granularity**. Since the technically, although not currently supported in CF, different collections with the same set of group metrics can be configured with different `granularity` I didn't add deduplication logic, to allow users to escape hatch this in the case  `granularity` or others values will be allowed in the future. 

If you are referring to same group metrics in the a single collection:

```
    asg.emitGroupMetrics(
      autoscaling.GroupMetric.PENDING_INSTANCES,
      autoscaling.GroupMetric.PENDING_INSTANCES,
    );
```

This is supported in CF and ASG, I will add a test case to make this more clear. Funnily enough this is why i have added the integ test (which I understand why it looks redundant), to verify all of this cases

Reply:
I think we should dedupe on our end even if it works in CF. We allow them to be specified in two ways (constructor & `emit` method), so it's only reasonable that we do some deduping and keep it clean to the user.

Consider that it the user accidentally repeats, it shouldn't trigger a new CF deployment. Feels like easy win for our customer.

Reply:
We can use a set for each collection to avoid duplication in a collection, but Im not sure we want to block users from a supported use case, i guess we can add it now and remove if needed in the future

Reply:
> Im not sure we want to block users from a supported use case

What's the use case? Specifying the value twice doesn't change any behaviour, right?

If we dedupe and sort, which should be very easy to implement for us, unneeded stack updates can be avoided (easy win!).

Reply:
replaced the inner array with a set which will deduplicate the group metrics